src/common/runner.cpp

```diff
@@ -2,19 +2,32 @@
 // Measurement-kit is free software. See AUTHORS and LICENSE for more
 // information on the copying conditions.
 
+#include <condition_variable>
 #include <measurement_kit/common.hpp>
+#include <mutex>
+#include <thread>
 
 namespace mk {
 
 Runner::Runner() {}
 
 void Runner::run_test(Var<NetTest> test, std::function<void(Var<NetTest>)> fn) {
     if (!running) {
+        std::mutex mutex;
+        std::unique_lock<std::mutex> lock(mutex);
+        std::condition_variable condition;
         // WARNING: below we're passing `this` to the thread, which means that
         // the destructor MUST wait the thread. Otherwise, when the thread dies
         // many strange things could happen (I have seen SIGABRT).
-        thread = std::thread([this]() { reactor->loop(); });
+        debug("runner: starting reactor in background...");
+        thread = std::thread([&condition, &lock, &mutex, this]() {
```

Comment:
Why are you passing the lock and the mutex to this callback? The documentation says that the lock does not need to be held for notification.